### PR TITLE
787 neighbor list popup

### LINKED ISSUE #787: Enable showing list of route paths that are using neighbor nodes (on hover).
On hover: 
List all routes (with routepaths) that use this node. Small button to open routepath to investigate them further.

### PR DESCRIPTION
**Pull request by kronken**
Closes #787 

Test by expanding route path and hover on neighbor nodes having a count higher than 0.
Click open route path by clicking on their ID.

Made a change in routePathService, before createRoutePath function took both routeId and routePath as parameters. However routeId is included in routePath, so I removed it.

### CODE REVIEW

File: src/components/map/layers/edit/RoutePathNeighborLinkLayer.tsx [hunk 1]
```diff
@@ -27,6 +32,39 @@ class RoutePathNeighborLinkLayer extends Component<IRoutePathLayerProps> {
             !== routePath!.routePathLinks!.filter(x => x.endNode.id === node.id).length;
     }
 
+    private nodeUsageView = (routePaths: IRoutePath[]) => {
```
Comment (Divinare) [inline src/components/map/layers/edit/RoutePathNeighborLinkLayer.tsx]: renderNodeUsageView

File: src/components/map/layers/edit/RoutePathNeighborLinkLayer.tsx [hunk 2]
```diff
@@ -27,6 +32,39 @@ class RoutePathNeighborLinkLayer extends Component<IRoutePathLayerProps> {
...
+        if (!routePaths || routePaths.length === 0) return;
+        return ReactDOMServer.renderToStaticMarkup(
+            <div className={s.usageList}>
```
Comment (Divinare) [inline src/components/map/layers/edit/RoutePathNeighborLinkLayer.tsx]: nit: nodeUsageList

File: src/components/map/layers/mapIcons/MarkerPopup.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                    if (parent === leafletMarker._popup._container) {
+                        return;
+                    }
+
+                    timeout && clearTimeout(timeout);
+                    timeout = setTimeout(
+                        () => {
+                            leafletMarker.openPopup();
+                        },
+                        500,
```
Comment (Divinare) [inline src/components/map/layers/mapIcons/MarkerPopup.ts]: Add this timeout constant to the start of file?

const OPEN_POPUP_TIMEOUT = 500;

File: src/components/map/layers/mapIcons/nodeMarker.scss [hunk 4]
```diff
@@ -46,3 +46,7 @@
     font-weight: 600;
     text-shadow: 0 0 5px white, 0 0 5px white;
 }
+
+.leaflet-popup-tip-container {
+    pointer-events: all !important;
```
Comment (Divinare) [inline src/components/map/layers/mapIcons/nodeMarker.scss]: Remove !important?

File: src/components/map/layers/mapIcons/NodeMarker.tsx [hunk 5]
```diff
@@ -41,6 +42,7 @@ interface INodeMarkerProps {
     onClickEventParams?: any;
     forcedVisibleNodeLabels?: NodeLabel[];
     markerClasses?: string[];
+    popupContent?: string;
```
Comment (Divinare) [inline src/components/map/layers/mapIcons/NodeMarker.tsx]: Add comment // static markup language (HTML)

File: src/components/map/layers/mapIcons/MarkerPopup.ts (hunk 3 above)
Comment (kronken) [reply src/components/map/layers/mapIcons/MarkerPopup.ts]: would OPEN_POPUP_DELAY be more declarative?

**Pull request merged**